agent/watch_handler.go

```diff
@@ -21,12 +21,26 @@ const (
 )
 
 // makeWatchHandler returns a handler for the given watch
-func makeWatchHandler(logOutput io.Writer, params interface{}) watch.HandlerFunc {
-	script := params.(string)
+func makeWatchHandler(logOutput io.Writer, handler interface{}) watch.HandlerFunc {
+	var args []string
+
+	// Figure out whether to run in shell or raw subprocess mode
```

Comment:
Ugh I'm sorry I just realized that `ExecScript` has special stuff on Windows (there's util_windows.go). I know there are some open issues for Windows that will likely be cleaned up by just using args, so we should probably keep using `ExecScript` in this path as well until we just deprecate it, that way we don't break more than we have to. Sorry to flip back on this.